src/encoded/static/components/visualizations.js

```diff
@@ -524,18 +531,28 @@ export class ChartList extends React.Component {
                             errorMessage = false;
                         }
                     }
-                    const barWidth = ((this.props.chartWidth - this.state.leftMargin) / this.state.chartMax) * this.state.chartData[d];
-                    const remainderWidth = this.props.chartWidth - barWidth - this.state.leftMargin;
+                    let barWidth = 0;
+                    let remainderWidth = 0;
+                    let leftMargin = 0;
+                    if (chartWidth > 500) {
+                        leftMargin = this.state.leftMargin;
+                        barWidth = ((chartWidth - leftMargin) / this.state.chartMax) * this.state.chartData[d];
+                        remainderWidth = chartWidth - barWidth - leftMargin;
+                    } else {
+                        leftMargin = '100%';
+                        barWidth = (chartWidth / this.state.chartMax) * this.state.chartData[d];
+                        remainderWidth = chartWidth - barWidth;
+                    }
```

Comment:
Takes some time for me to figure out what this if/else is for.

Reply:
Should I add a comment? I didn't because it was in render() but I could clarify somewhere.

Reply:
It’s up to you, but it just looked difficult for someone else to maintain it later. It’s in a render() return, but it’s outside of JSX, so comments aren’t awkward, in my view. Comments in JSX are kind of awkward, so I haven’t done that very much.